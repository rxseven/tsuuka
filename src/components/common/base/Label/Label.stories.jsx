import React from 'react';
import { storiesOf } from '@storybook/react';

import Label from './index';

const title = 'Components/Common/Base/Label';
const props = { children: 'Form label', htmlFor: 'username' };

storiesOf(title, module)
  .addParameters({
    info: {
      text: `
  **Usage:** for more information, see [Bootstrap - Forms](https://getbootstrap.com/docs/4.1/components/forms/).
`
    }
  })
  .add('default', () => <Label {...props} />)
  .add('small', () => <Label {...props} small />);
