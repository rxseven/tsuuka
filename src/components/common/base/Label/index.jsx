import PropTypes from 'prop-types';
import exact from 'prop-types-exact';
import React from 'react';
import styled from 'styled-components';

const Tag = styled.label`
  color: ${({ small, theme }) =>
    small ? theme.color.secondary : theme.color.base};
  font-size: ${({ small }) => (small ? '0.75rem' : 'inherit')};
  margin-bottom: ${({ small }) => (small ? '0.35rem' : 'inherit')};
`;

const propTypes = exact({
  children: PropTypes.string.isRequired,
  className: PropTypes.string,
  htmlFor: PropTypes.string.isRequired,
  small: PropTypes.bool
});

const defaultProps = {
  className: '',
  small: false
};

function Label({ children, ...props }) {
  return <Tag {...props}>{children}</Tag>;
}

Label.propTypes = propTypes;
Label.defaultProps = defaultProps;

export default Label;
