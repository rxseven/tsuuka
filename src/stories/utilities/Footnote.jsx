import React from 'react';
import styled from 'styled-components';

const Frame = styled.div`
  bottom: 2rem;
  font-size: 14px;
  position: fixed;
`;

function Footnote({ children }) {
  return <Frame>{children}</Frame>;
}

export default Footnote;
