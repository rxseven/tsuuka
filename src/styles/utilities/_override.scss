// Reset button's default styles
button:focus {
  box-shadow: none !important;
  outline: 0 !important;
}
