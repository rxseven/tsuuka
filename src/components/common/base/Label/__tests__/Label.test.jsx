import { factory } from 'tests/utilities';
import Label from '../index';

// Arrange
const seed = { content: 'content' };
const source = {
  children: seed.content,
  htmlFor: 'email'
};
const input = { ...seed, ...source };

// Setup
function setup(props) {
  return factory(Label, source, props);
}

// Test suites
describe('<Label />', () => {
  it('should render without crashing', () => {
    setup();
  });

  it('should render passed children correctly', () => {
    const expected = { content: input.content };
    const { component } = setup();

    expect(component).toHaveTextContent(expected.content);
  });
});
