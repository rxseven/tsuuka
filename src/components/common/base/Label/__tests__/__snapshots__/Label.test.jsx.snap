// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label /> Snapshot tests should render correctly 1`] = `
<label
  className="sc-bdVaJa dekHzo"
  htmlFor="username"
>
  content
</label>
`;
